autograph_utils/__init__.py

```diff
@@ -95,6 +128,19 @@ def detail(self):
         return f"Certificate expired in the past on {self.not_after}"
 
 
+class CertificateHasWrongSubject(BadCertificate):
+    def __init__(self, actual, check):
+        self.check = check
```

Comment:
Maybe pass description as string here instead of check?  (IDK, maybe matter of taste)